lib/c-api/src/wasm_c_api/engine.rs

```diff
@@ -84,6 +89,7 @@ pub struct wasm_config_t {
     pub(super) middlewares: Vec<wasmer_middleware_t>,
     pub(super) nan_canonicalization: bool,
     pub(super) features: Option<Box<wasmer_features_t>>,
+    #[cfg(feature = "compiler")]
```

Comment:
We should do a breaking change and unite all the sys features into one field (for sys).

Those are: middleware/nan_canonicalization/target